packages/zendesk-adapter/src/filters/ticket_form.ts

```diff
@@ -47,27 +53,59 @@ type TicketForm = {
   }[]
 }
 
+const isCustomStatusesEnabled = async (elementSource?: ReadOnlyElementsSource): Promise<boolean> => {
+  if (elementSource === undefined) {
+    log.error('Returning that customStatuses is disabled since no element source was provided')
+    return false
+  }
+
+  const accountFeatures = await elementSource.get(
+    new ElemID(
+      ZENDESK,
+      ACCOUNT_FEATURES_TYPE_NAME,
+      'instance',
+      ElemID.CONFIG_NAME
+    )
+  )
+
+  if (!isInstanceElement(accountFeatures)) {
+    log.warn('Returning that customStatuses is disabled since accountFeatures is not an instance')
+    return false
+  }
+  const customStatusesEnabled = accountFeatures.value.custom_statuses_enabled
+  if (customStatusesEnabled === undefined) {
+    log.warn('Returning that customStatuses is disabled since custom_statuses_enabled does not exist')
+    return false
+  }
+  return accountFeatures.value.custom_statuses_enabled.enabled
+}
+
 /**
- * checks if both statuses and custom_statuses are defined
+ * checks if both statuses and custom_statuses are defined and if custom_statuses is enabled
  */
-const isInvalidTicketForm = (instanceValue: Record<string, unknown>): instanceValue is TicketForm =>
-  _.isArray(instanceValue.agent_conditions)
-  && instanceValue.agent_conditions.some(condition =>
-    _.isArray(condition.child_fields)
+const isInvalidTicketForm = (instanceValue: Record<string, unknown>, hasCustomStatusesEnabled: boolean)
+  : instanceValue is TicketForm =>
```

Comment:
readability nit: maybe
```suggestion
const isInvalidTicketForm = (
  instanceValue: Record<string, unknown>,
  hasCustomStatusesEnabled: boolean,
): instanceValue is TicketForm =>
```